#!/usr/bin/env python
PACKAGE = "turtlebot_trajectory_testing"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("c_p", double_t, 0, "Proportional gain",    1, 0)
gen.add("c_d", double_t, 0, "Derivative gain",    5, 0)
gen.add("c_lambda", double_t, 0, "Lambda gain",    1, 0)
gen.add("epsilon", double_t, 0, "Epsilon",    .01, 0)

gen.add("v_max", double_t, 0, "Maximum linear velocity",    0.5, 0)
gen.add("w_max", double_t, 0, "Maximum angular velocity",    4, 0)
gen.add("a_max", double_t, 0, "Maximum linear acceleration",    .55, 0)
gen.add("w_dot_max", double_t, 0, "Maximum angular acceleration",    1.78, 0)

exit(gen.generate(PACKAGE, "turtlebot_trajectory_testing", "NIConfigUtility"))
